docs/docsite/rst/lookup_guide.rst

```diff
@@ -0,0 +1,159 @@
+.. _ansible_collections.community.hashi_vault.docsite.lookup_guide:
+
+************
+Lookup guide
+************
+
+This guide is not a comprehensive listing of included lookup plugins and how to use them, rather it is intended to explain the role of the lookup plugins in this collection and how they are they used, especially when compared to modules of the same name.
+
+.. contents::
+  :local:
+  :depth: 2
+
+
+About the ``hashi_vault`` lookup
+================================
+
+Before this collection existed, the ``hashi_vault`` lookup plugin was the only Vault-related content included in Ansible. Eventually it was moved to :ref:`community.general <plugins_in_community.general>` before being split out into this collection, where it was the only content in the collection for a long time. As a result, it's the most used plugin for Vault, and the one most people are familiar with.
+
+``hashi_vault`` lookup considerations
+-------------------------------------
+
+Due to the history of the ``hashi_vault`` lookup's development, and it being the only one around, it kind of does many jobs, making it versatile, but sometimes unintuitive.
+
+First, the lookup does authentication, taking parameters for various login types, performing a login, and acquiring a token with which it can make additional calls to Vault.
+
+Second, the lookup does a generic read. That makes it very useful because it can read any kind of Vault path, without having to be written with that type of path in mind.
+
+It's worth understanding that by far the most common use case is reading secrets, and by far the most common secret store is the ``kv`` (key/value) store that's built into Vault. Most implementations are using v2 of the ``kv`` store.
+
+So the third thing the lookup does is assume that you're probably trying to read a ``kv`` secret, and to then try to infer if the response is from ``kv2``, because the responses from version 2 include metadata and have the secret value additionally wrappped in another structure. In that way, it seeks to make the response from version 2 look more like the response from version 1.
```

Comment:
I'd probably shorten this and turn it into a list:
```suggestion
The ``hashi_vault`` lookup plugin performs three main tasks:

 - authentication, taking parameters for various login types, performing a login, and acquiring a token with which it can make additional calls to Vault.
- a generic read operation, which allows it to read any kind of Vault path, without having to be written with that type of path in mind.
- reading secrets.

Reading secrets is the most common use case, with the ``kv`` (key/value) store built into Vault as by far the most common secret store. Most implementations use v2 of the ``kv`` store. To make reading v2 ``kv`` secrets easy, the lookup plugin assumes that you're probably trying to read a ``kv`` secret, and tries to infer if the response is from ``kv2``, because the responses from version 2 include metadata and have the secret value additionally wrapped in another structure. The lookup plugin seeks to make ``kv2`` responses look more like responses from version 1.
```

Reply:
this is much better thanks, I will have to reword slightly but I'll commit and edit after